orders/src/main/resources/static/js/orders.js

```diff
@@ -0,0 +1,151 @@
+window.productsByName = new Map();
+window.productsByBarcode = new Map();
+window.productsByCategory = new Map();
+window.finalOrders = new Set();
+
+function removeOptions(selectbox) {
+    var i;
+    for (i = selectbox.options.length - 1; i >= 0; i--) {
+        selectbox.remove(i);
+    }
+}
+
+//  initPage
+window.barcode_input = document.getElementById("enterBarcode");
+window.barcode_input.disabled = true;
+window.orderTable = document.getElementById("orderTable");
+window.orderSummaryTable = document.getElementById("orderSummaryTable").getElementsByTagName("tbody")[0];
+window.categories = new Set();
+window.selectDepartmentDropdown = document.getElementById("selectDepartment");
+window.chooseProductName = document.getElementById("chooseProductName");
+window.enterAmount = document.getElementById("enterAmount");
+let i;
+for (i = 0; i < window.products.length; i++) {
+    window.categories = window.categories.add(window.products[i].category)
+}
+for (let val of window.categories) {
+    window.productsByCategory.set(val, new Set());
+    let option = document.createElement('option');
+    option.text = val;
+    window.selectDepartmentDropdown.appendChild(option);
+}
+for (let product of window.products) {
+    window.productsByCategory.get(product.category).add(product);
+    window.productsByName.set(product.name, product);
+    window.productsByBarcode.set(product.barcode, product);
+}
+
+function updateOrderTableInfoFromProduct(product) {
+    let row = window.orderTable.rows[window.orderTable.rows.length - 1];
+    this.selectDepartmentDropdown.value = product.category;
+    let option = document.createElement('option');
+    option.text = product.name;
+    window.chooseProductName.appendChild(option);
+    row.cells[3].innerText = product.price;
+    window.barcode_input.value = product.barcode;
+    row.cells[5].innerText = "0";
+}
+
+function clearOrderTableExceptCategory() {
```

Comment:
why is this clearing the order? why not name the function "clearProductFromOrderForm" or something like that? or simply "clearMalformedProduct"

Reply:
because the table that is used to choose the product is called "orderTable" and we are clearing it except for the category. If you still think we should rename, fine - tell me.

Reply:
simple name is usually better - choose whichever name you think is easier to understand